Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/Parameters/ObjectIDDataParameterDefinitionTest.cs

```diff
@@ -0,0 +1,178 @@
+{
+  [Test]
+  public void GetParameterValue ()
+  {
+    var storageTypeInformationMock = new Mock<IStorageTypeInformation>();
+    var dummyValue = DomainObjectIDs.Order1;
+
+    var dummyGuidResult = Guid.NewGuid();
+
+    storageTypeInformationMock
+        .Setup(_ => _.ConvertToStorageType(dummyValue.Value))
+        .Returns(dummyGuidResult);
+    storageTypeInformationMock.Setup(_ => _.DotNetType).Returns(typeof(Guid));
+
+    var objectIDDataParameterDefinition = new ObjectIDDataParameterDefinition(storageTypeInformationMock.Object);
+
+    var result = objectIDDataParameterDefinition.GetParameterValue(dummyValue);
+
+    Assert.That(result, Is.EqualTo(dummyGuidResult));
+    storageTypeInformationMock.Verify();
+  }
+
+  [Test]
+  public void GetParameterValue_WithNull ()
+  {
+    var storageTypeInformationStub = new Mock<IStorageTypeInformation>();
+    object dummyValue = null;
+
+    storageTypeInformationStub
+        .Setup(_ => _.ConvertToStorageType(dummyValue))
+        .Returns(DBNull.Value);
+    storageTypeInformationStub.Setup(_ => _.DotNetType).Returns(typeof(Guid));
+
+    var objectIDDataParameterDefinition = new ObjectIDDataParameterDefinition(storageTypeInformationStub.Object);
+
+    var result = objectIDDataParameterDefinition.GetParameterValue(dummyValue);
+
+    Assert.That(result, Is.EqualTo(DBNull.Value));
+  }
+
+  [Test]
+  public void GetParameterValue_WithInvalidValue_ThrowsArgumentException ()
+  {
+    var storageTypeInformationStub = new Mock<IStorageTypeInformation>();
+    var dummyValue = new ObjectID(typeof(Customer), Guid.NewGuid());
+
+    storageTypeInformationStub.Setup(_ => _.DotNetType).Returns(typeof(int));
+
+    var objectIDDataParameterDefinition = new ObjectIDDataParameterDefinition(storageTypeInformationStub.Object);
+
+    Assert.That(
+        () => objectIDDataParameterDefinition.GetParameterValue(dummyValue),
+        Throws.InstanceOf<ArgumentException>().With.ArgumentExceptionMessageEqualTo(
+            string.Format("Parameter '{0}' has type '{2}' when type '{1}' was expected.", "parameterObject.Value", typeof(int), typeof(Guid)),
+            "parameterObject.Value"));
+  }
+
+
+  [Test]
+  public void CreateDataParameter_WithIntValue ()
+  {
+    var parameterValue = 42;
+    var commandMock = new Mock<IDbCommand>(MockBehavior.Strict);
+    var dataParameterMock = new Mock<IDbDataParameter>(MockBehavior.Strict);
+
+    var storageTypeInformation = new StorageTypeInformation(
+        typeof(int),
+        "test",
+        DbType.Int32,
+        false,
+        null,
+        typeof(int),
+        new DefaultConverter(typeof(int)));
+
+    commandMock.Setup(mock => mock.CreateParameter()).Returns(dataParameterMock.Object).Verifiable();
+
+    dataParameterMock.SetupSet(mock => mock.DbType = storageTypeInformation.StorageDbType).Verifiable();
+    dataParameterMock.SetupSet(mock => mock.Value = parameterValue).Verifiable();
```

Comment:
for these, we actually want to test if the correct properties where set. I would suggest using verify calls in the assert block.

I do like uto set up my verify when doing method calls or mixed stuff. when having an object that simply gets proeprties assigned, the assert block feels more natural.

this will affect the other tests too, here and in SimpleDataParamDefTest.

Reply:
Update: I have seen you're using strict mock for the parameter. okay, then you need to use this syntax. not 100% sure if it's the best way to test this / overkill. might very well be, though.

We also should consider that we will eventually change the parameter to a SqlDataParameter (at least for the SimpleDataParamDefTest) whch will change the way we have to do the assertions.